### PR TITLE
BoatAngularGenerator: add dependency versions for ang@13

### PR DESCRIPTION
**Pull request by brianvabb**
### Description

BoatAngularGenerator: add dependency versions to support `angular@13`. The compatible peer dependency versions were found in the following resources:

- **RxJS version**: https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3
- **TypeScript version**: https://github.com/angular/angular/releases?expanded=true&page=5&q=13#:~:text=TypeScript%20versions%20older%20than%204.4.2%20are%20no%20longer%20supported. 
- **NG-PACKAGR version**: https://github.com/ng-packagr/ng-packagr/blob/1a503338c623ad1dfefc379067b8637daa19f901/package.json#L3-L71

Fixes # nojira

#### Type of change

- [X] Versions (update of Angular dependency versions)

### CODE REVIEW

**Comment by brianvabb** [comment]
- [ ] Needs review before merge.

**Review by santam85 — APPROVED** [review]

**Comment by ngohungphuc** [comment]
@brianvabb what about zonejs version ? I think we should use 0.11.4

image

File: boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java [hunk 1]
```diff
@@ -222,14 +222,20 @@ private void addDependencies(SemVer angularVersion) {
             supportingFiles.add(new SupportingFile("ng-package.mustache", getIndexDirectory(), "ng-package.json"));
             supportingFiles.add(new SupportingFile("tsconfig.mustache", getIndexDirectory(), "tsconfig.json"));
             additionalProperties.put("zonejsVersion", "0.10.2");
```
Comment (ngohungphuc) [inline boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java:224]: @santam85 can we just use 1 version of rxjs which is in this case 0.11.7 ?

Comment (santam85) [reply boat-scaffold/src/main/java/com/backbase/oss/codegen/angular/BoatAngularGenerator.java:224]: I guess you mean zone.js? let's use the default version that gets generated with `ng new` for each Angular major, just to be sure...

**Comment by santam85** [comment]
- [ ] fix code smells

**Review by daiscog — APPROVED** [review]
Is zonejsVersion needed at all?  It doesn't appear in the generated peerDependencies.  I guess it's implied via the Angular dependency:

```
  "peerDependencies": {
    "@angular/core": ">=11.0.0",
    "@backbase/foundation-ang": ">=6.15.1",
    "rxjs": ">=6.6.0"
  },
```

**Comment by ngohungphuc** [comment]
@santam85 @daiscog Fixed the build. Please check it again then I will merge it.

**Review by ngohungphuc — APPROVED** [review]

**Review by daiscog — APPROVED** [review]

**Pull request merged**